irksome/galerkin_stepper.py

```diff
@@ -1,48 +1,45 @@
-from FIAT import (Bernstein, DiscontinuousElement, DiscontinuousLagrange,
-                  IntegratedLegendre, Lagrange, Legendre)
+from FIAT import (Bernstein, DiscontinuousLagrange,
+                  IntegratedLegendre, Lagrange,
+                  NodalEnrichedElement, RestrictedElement)
 from ufl.constantvalue import as_ufl
 from .base_time_stepper import StageCoupledTimeStepper
 from .bcs import bc2space, stage2spaces4bc
 from .deriv import TimeDerivative, expand_time_derivatives
 from .labeling import split_quadrature
 from .scheme import create_time_quadrature, ufc_line
 from .tools import dot, reshape, replace, vecconst
+from .discontinuous_galerkin_stepper import getElement as getTestElement
 import numpy as np
 from firedrake import TestFunction, Constant
 
 
 def getTrialElement(basis_type, order):
-    if basis_type == "Bernstein":
+    if basis_type is not None:
+        basis_type = basis_type.lower()
+    if basis_type == "bernstein":
         return Bernstein(ufc_line, order)
     elif basis_type == "integral":
         return IntegratedLegendre(ufc_line, order)
     else:
         # Let recursivenodes handle the general case
-        variant = None if basis_type == "Lagrange" else basis_type
+        variant = None if basis_type == "lagrange" else basis_type
         return Lagrange(ufc_line, order, variant=variant)
 
 
-def getTestElement(basis_type, order):
-    if order == 0:
-        return DiscontinuousLagrange(ufc_line, order)
-    elif basis_type == "Bernstein":
-        return DiscontinuousElement(Bernstein(ufc_line, order))
-    elif basis_type == "integral":
-        return Legendre(ufc_line, order)
-    else:
-        # Let recursivenodes handle the general case
-        variant = None if basis_type == "Lagrange" else basis_type
-        return DiscontinuousLagrange(ufc_line, order, variant=variant)
-
-
 def getElements(basis_type, order):
     if isinstance(basis_type, (tuple, list)):
         trial_type, test_type = basis_type
     else:
         trial_type = basis_type
         test_type = basis_type
-    L_trial = getTrialElement(trial_type, order)
+
     L_test = getTestElement(test_type, order-1)
+    if trial_type == "enriched":
```

Comment:
Not sure "enriched" is how I would name this mathematically absent the technical naming within FIAT, but I really don't have a better idea at the present time.